pkg/internal/checkup/executor/executor.go

```diff
@@ -257,34 +261,70 @@ func extractSectionStatistics(input, sectionStart, sectionEnd string) (string, e
 	return strings.Join(lines[startLineIdx+1:endLineIdx], "\n"), nil
 }
 
-func parseTestpmdStats(input string) (map[string]int64, error) {
-	params := make(map[string]int64)
-	scanner := bufio.NewScanner(strings.NewReader(input))
-	for scanner.Scan() {
-		line := scanner.Text()
-		if line == "" {
-			continue
+func parseTestpmdStats(input string) ([]testPmdPortStats, error) {
+	statistics := make([]testPmdPortStats, testPmdPortStatsSize)
+	const (
+		port0SectionStart   = "Forward statistics for port 0"
+		port0SectionEnd     = "----------------------------------------------------------------------------"
+		port1SectionStart   = "Forward statistics for port 1"
+		port1SectionEnd     = port0SectionEnd
+		SummarySectionStart = "Accumulated forward statistics for all ports+"
+		SummarySectionEnd   = "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++"
+	)
+
+	statsIdxToStartSectionMap := map[testPmdStatsIdx]string{
+		testPmdStatsPort0:       port0SectionStart,
+		testPmdStatsPort1:       port1SectionStart,
+		testPmdPortStatsSummary: SummarySectionStart,
+	}
+	statsIdxToEndtSectionMap := map[testPmdStatsIdx]string{
+		testPmdStatsPort0:       port0SectionEnd,
+		testPmdStatsPort1:       port1SectionEnd,
+		testPmdPortStatsSummary: SummarySectionEnd,
+	}
+
+	for statsIdx := testPmdStatsPort0; statsIdx < testPmdPortStatsSize; statsIdx++ {
+		sectionString, err := extractSectionStatistics(input, statsIdxToStartSectionMap[statsIdx], statsIdxToEndtSectionMap[statsIdx])
+		if err != nil {
+			return nil, fmt.Errorf("failed parsing section on port %d: %w", statsIdx, err)
+		}
+		err = parseTestpmdStatsSection(&statistics[statsIdx], sectionString)
+		if err != nil {
+			return nil, err
 		}
-		keyValuePairs := strings.Split(line, "  ")
-		for _, pair := range keyValuePairs {
-			if pair == "" {
-				continue
-			}
-			fieldsWithoutDuplicateSpaces := strings.Fields(pair)
-			formattedKeyValuePairString := strings.Join(fieldsWithoutDuplicateSpaces, " ")
-			parts := strings.Split(formattedKeyValuePairString, ": ")
-
-			key := strings.TrimSpace(parts[0])
-			valueString := strings.TrimSpace(parts[1])
-
-			value, err := strconv.ParseInt(valueString, 10, 64)
-			if err != nil {
-				return nil, err
-			}
-			params[key] = value
+	}
+
+	return statistics, nil
+}
+
+func parseTestpmdStatsSection(stats *testPmdPortStats, section string) error {
+	const (
+		RXPacketsIndex = 1
+		RXDroppedIndex = 3
+		RXTotalIndex   = 5
+		TXPacketsIndex = 1
+		TXDroppedIndex = 3
+		TXTotalIndex   = 5
+	)
+	lines := strings.Split(section, "\n")
+	for i := 0; i < len(lines); i++ {
```

Comment:
this can be a simple `for i := range lines`

Reply:
totally :) DONE